tests/cmds/test_wallet.py

```diff
@@ -160,3 +188,854 @@ async def get_coin_records(self, request: GetCoinRecords) -> Dict[str, Any]:
+    test_rpc_clients.wallet_rpc_client = inst_rpc_client
+    command_args = ["wallet", "take_offer", test_offer_file_name, "--no-confirm", FINGERPRINT_ARG, "-m1", "--reuse"]
+    success, output = run_cli_command(capsys, root_dir, command_args)
+    assert success
+    # these are various things that should be in the output
+    cat1 = bytes32.from_hexstr("fd6a341ed39c05c31157d5bfea395a0e142398ced24deea1e82f836d7ec2909c")
+    cat2 = bytes32.from_hexstr("dc59bcd60ce5fc9c93a5d3b11875486b03efb53a53da61e453f5cf61a7746860")
+    assert_list = [
+        "  OFFERED:\n"
+        "    - XCH (Wallet ID: 1): 10.0 (10000000000000 mojos)\n"
+        f"    - {cat1.hex()}: 100.0 (100000 mojos)",
+        "  REQUESTED:\n"
+        f"    - {cat2.hex()}: 10.0 (10000 mojos)\n"
+        "    - accce8e1c71b56624f2ecaeff5af57eac41365080449904d0717bd333c04806d: 0.001 (1 mojo)",
+        "Accepted offer with ID dfb7e8643376820ec995b0bcdb3fc1f764c16b814df5e074631263fcf1e00839",
+    ]
+    cli_assert_shortcut(output, assert_list)
+    expected_calls: logType = {
+        "cat_asset_id_to_name": [
+            (cat1,),
+            (cat2,),
+            (bytes32.from_hexstr("accce8e1c71b56624f2ecaeff5af57eac41365080449904d0717bd333c04806d"),),
+        ],
+        "take_offer": [(Offer.from_bech32(test_offer_file_bech32), None, 1000000000000, uint64(0), None)],
+    }
+    test_rpc_clients.wallet_rpc_client.check_log(expected_calls)
+
+
+def test_cancel_offer(capsys: object, get_test_cli_clients: Tuple[TestRpcClients, Path]) -> None:
+    test_rpc_clients, root_dir = get_test_cli_clients
+
+    # set RPC Client
+    class CancelOfferRpcClient(TestWalletRpcClient):
+        async def get_offer(self, trade_id: bytes32, file_contents: bool = False) -> TradeRecord:
+            self.add_to_log("get_offer", (trade_id, file_contents))
+            offer = Offer.from_bech32(test_offer_file_bech32)
+            return TradeRecord(
+                confirmed_at_index=uint32(0),
+                accepted_at_time=uint64(0),
+                created_at_time=uint64(12345678),
+                is_my_offer=True,
+                sent=uint32(0),
+                sent_to=[],
+                offer=bytes(offer),
+                taken_offer=None,
+                coins_of_interest=offer.get_involved_coins(),
+                trade_id=offer.name(),
+                status=uint32(TradeStatus.PENDING_ACCEPT.value),
+            )
+
+        async def cancel_offer(self, trade_id: bytes32, fee: uint64 = uint64(0), secure: bool = True) -> None:
+            self.add_to_log("cancel_offer", (trade_id, fee, secure))
+            return None
+
+    inst_rpc_client = CancelOfferRpcClient()  # pylint: disable=no-value-for-parameter
+    test_rpc_clients.wallet_rpc_client = inst_rpc_client
+    command_args = ["wallet", "cancel_offer", FINGERPRINT_ARG, "--no-confirm", "-m1", "--id", test_offer_id]
+    success, output = run_cli_command(capsys, root_dir, command_args)
+    assert success
```

Comment:
if every invocation of `run_cli_command()` asserts `success`, you could probably just move that into the function and not return it

Reply:
yeah i ended up moving it